ddl/get.lua

```diff
@@ -52,12 +52,51 @@ local function _get_index(box_space, box_index)
     return ddl_index
 end
 
-local function get_space_sharding_key(space_name)
-    if box.space._ddl_sharding_key == nil then
+local metadata_cache = {}
+setmetatable(metadata_cache, {__mode = 'k'})
+
+local function get_metadata(space_name, metadata_name)
+    local metadata_space_name = string.format("_ddl_%s", metadata_name)
+
+    local box_space = box.space[space_name]
+    if box_space == nil then
+        return nil
+    end
+
+    local index = box_space.index[0]
+
+    if index and metadata_cache[index] and metadata_cache[index][metadata_name] then
+        return metadata_cache[index][metadata_name]
+    end
+
+    if box.space[metadata_space_name] == nil then
+        return nil
+    end
+
+    local record = box.space[metadata_space_name]:get{space_name}
+    if index then
+        metadata_cache[index] = metadata_cache[index] or {}
+        metadata_cache[index][metadata_name] = record
+    end
+
+    return record
+end
```

Comment:
I vote for don't do caching here. There are several thoughts around:

1. Now the module constructs the schema each time it is requested. So frequent access assumes caller side caching anyway. If we'll implement caching within the module (as proposed in #69), we'll likely do it on a higher level.
2. There are several thoughts why this caching algorithm may not work as expected:
    - It depends on when the garbage collector will delete an index object. So even if the space is deleted and created again we can return data from the cache.
    - Even if we would call `collectgarbage()` manually (that would be strange but anyway), it does not protect us from the situation, when a user kept a link to the old index object.
    - The modification of the space (inserting, removing, updating tuples) will not loss a reference to the index object. So, again, we can return obsolete data.

Reply:
Metadata cache has been removed